recipe/session/sessmodels/models.go

```diff
@@ -147,16 +153,16 @@ type NormalisedErrorHandlers struct {
 }
 
 type SessionContainer struct {
-	RevokeSession            func() error
-	GetSessionData           func() (map[string]interface{}, error)
-	UpdateSessionData        func(newSessionData map[string]interface{}) error
-	GetUserID                func() string
-	GetAccessTokenPayload    func() map[string]interface{}
-	GetHandle                func() string
-	GetAccessToken           func() string
-	UpdateAccessTokenPayload func(newAccessTokenPayload map[string]interface{}) error
-	GetTimeCreated           func() (uint64, error)
-	GetExpiry                func() (uint64, error)
+	RevokeSession            func(userContext supertokens.UserContext) error
+	GetSessionData           func(userContext supertokens.UserContext) (map[string]interface{}, error)
+	UpdateSessionData        func(newSessionData map[string]interface{}, userContext supertokens.UserContext) error
+	GetUserID                func(userContext supertokens.UserContext) string
+	GetAccessTokenPayload    func(userContext supertokens.UserContext) map[string]interface{}
+	GetHandle                func(userContext supertokens.UserContext) string
+	GetAccessToken           func(userContext supertokens.UserContext) string
+	UpdateAccessTokenPayload func(newAccessTokenPayload map[string]interface{}, userContext supertokens.UserContext) error
+	GetTimeCreated           func(userContext supertokens.UserContext) (uint64, error)
+	GetExpiry                func(userContext supertokens.UserContext) (uint64, error)
```

Comment:
Perhaps keep the original functions, but add a version of them with context. So for ex, you have `RevokeSession`, but then you will also have `RevokeSessionWithContext` and so non..

Implementation wise, `RevokeSession` will always just called `RevokeSessionWithContext` with an empty context object, and internally, we will never call `RevokeSession`, but always call the with context versions.